FWCore/ParameterSet/bin/edmWriteConfigs.cpp

```diff
@@ -136,18 +138,33 @@ namespace {
     }
   }
 
-  void writeModulesFile() {
+  bool open_temp(std::string& path, std::ofstream& f) {
+    path += "/XXXXXX";
+    path += '\0';  // ensure that the string is null-terminated
+    int fd = mkstemp(path.data());
+    if (fd != -1) {
+      f.open(path.c_str(), std::ios_base::trunc | std::ios_base::out);
+      close(fd);
+      return true;
+    }
+    return false;
+  }
+
+  bool writeModulesFile() {
     if (std::filesystem::exists(std::filesystem::current_path() / "modules.py"))
       return;
-    std::array<char, L_tmpnam> buffer;
-    std::tmpnam(buffer.data());
-    std::ofstream file{buffer.data()};
-
-    file << "from FWCore.ParameterSet.ModulesProxy import _setupProxies\n"
-            "locals().update(_setupProxies(__file__))\n";
-    file.close();
-    std::filesystem::copy(buffer.data(), "modules.py");
-    std::filesystem::remove(buffer.data());
+    std::ofstream file;
+    std::string path(std::filesystem::current_path());
+    bool res = open_temp(path, file);
+    if (res) {
+      file << "from FWCore.ParameterSet.ModulesProxy import _setupProxies\n"
+              "locals().update(_setupProxies(__file__))\n";
+      file.close();
+      std::filesystem::copy(path.data(), "modules.py");
+      std::filesystem::remove(path.data());
+      return true;
+    }
+    return false;
```

Comment:
Now this could become
```suggestion
    auto const path = makeUniqueFilename(std::filesystem::current_path());
    std::ofstream file{path.c_str());
    file << "from FWCore.ParameterSet.ModulesProxy import _setupProxies\n"
            "locals().update(_setupProxies(__file__))\n";
    file.close();
    std::filesystem::copy(path.c_str(), "modules.py");
    std::filesystem::remove(path.c_str());
```